File: lib/layer2_core/light/e_on_primary_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnPrimaryLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl17;
      case 'standard': return cstl17;
      case 'blue': return cbl17;
      case 'green': return cgl17;
      case 'red': return crl17;
      case 'nothing': return cnol17;
      default: return cstl17;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnPrimaryLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnPrimaryLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl17
            AppTheme.STANDARD -> R.color.cstl17
            AppTheme.BLUE -> R.color.cbl17
            AppTheme.GREEN -> R.color.cgl17
            AppTheme.RED -> R.color.crl17
            AppTheme.NOTHING -> R.color.cnol17
        }
    }
}
